src/invidious/helpers/youtube_api.cr

```diff
@@ -2,120 +2,200 @@
+        "gl"            => region || "US", # Can't be empty!
+        "clientName"    => "WEB",
+        "clientVersion" => HARDCODED_CLIENT_VERS,
+      },
+    }
   end
 
-  return _youtube_api_post_json("/youtubei/v1/browse", data)
-end
+  ####################################################################
+  # browse(continuation)
+  # browse(browse_id, params)
+  # browse(browse_id, params, region)
+  #
+  # Requests the youtubei/v1/browse endpoint with the required headers
+  # and POST data in order to get a JSON reply in english that can
+  # be easily parsed.
+  #
+  # A region can be provided, default is US.
+  #
+  # The requested data can either be:
+  #
+  #  - A continuation token (ctoken). Depending on this token's
+  #    contents, the returned data can be playlist videos, channel
+  #    community tab content, channel info, ...
+  #
+  #  - A playlist ID (parameters MUST be an empty string)
+  #
+  def browse(continuation : String)
+    # JSON Request data, required by the API
+    data = {
+      "context"      => self.make_context("US"),
+      "continuation" => continuation,
+    }
 
-####################################################################
-# request_youtube_api_search(search_query, params, region)
-#
-# Requests the youtubei/v1/search endpoint with the required headers
-# and POST data in order to get a JSON reply. As the search results
-# vary depending on the region, a region code can be specified in
-# order to get non-US results.
-#
-# The requested data is a search string, with some additional
-# paramters, formatted as a base64 string.
-#
-def request_youtube_api_search(search_query : String, params : String, region = nil)
-  # JSON Request data, required by the API
-  data = {
-    "query"   => search_query,
-    "context" => make_youtube_api_context(region),
-    "params"  => params,
-  }
-
-  return _youtube_api_post_json("/youtubei/v1/search", data)
-end
-
-####################################################################
-# _youtube_api_post_json(endpoint, data)
-#
-# Internal function that does the actual request to youtube servers
-# and handles errors.
-#
-# The requested data is an endpoint (URL without the domain part)
-# and the data as a Hash object.
-#
-def _youtube_api_post_json(endpoint, data)
-  # Send the POST request and parse result
-  response = YT_POOL.client &.post(
-    "#{endpoint}?key=#{HARDCODED_API_KEY}",
-    headers: HTTP::Headers{"content-type" => "application/json; charset=UTF-8"},
-    body: data.to_json
-  )
-
-  initial_data = JSON.parse(response.body).as_h
-
-  # Error handling
-  if initial_data.has_key?("error")
-    code = initial_data["error"]["code"]
-    message = initial_data["error"]["message"].to_s.sub(/(\\n)+\^$/, "")
-
-    raise InfoException.new("Could not extract JSON. Youtube API returned \
-      error #{code} with message:<br>\"#{message}\"")
+    return self._post_json("/youtubei/v1/browse", data)
+  end
+
+  # :ditto:
```

Comment:
The ditto doc directive has been deprecated and is removed in Crystal 1.1.0.

Reply:
How weird, this is still in their reference documentation: https://crystal-lang.org/reference/syntax_and_semantics/documenting_code.html#ditto

Reply:
Yeah I don't think that got updated along with the removal of `:ditto:` in crystal itself. Hell, I think `:ditto:` is still used in multiple places within crystal's source as well.

But either way, [it's been removed.](https://github.com/crystal-lang/crystal/pull/10755)

Reply:
In continuation to my comment from the other review, all of the doc directives are really just annotations to `crystal docs` anyways. If we decide to not support it then that's another reason why this`:ditto` should be removed.

Reply:
Watch out, they removed `ditto`, not `:ditto:`!
https://github.com/crystal-lang/crystal/pull/10755/files

Reply:
Well... I'm an  idiot.